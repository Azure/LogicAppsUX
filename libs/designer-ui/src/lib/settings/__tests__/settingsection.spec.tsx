import { SettingsSection } from '..';
import type { SettingSectionComponentProps } from '..';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';

describe('ui/settings/settingsection', () => {
  let minimal: SettingSectionComponentProps, renderer: ReactShallowRenderer.ShallowRenderer;

  beforeEach(() => {
    minimal = {
      id: 'settingId',
      title: 'random setting',
      expanded: false,
      isInverted: false,
      isReadOnly: false,
      renderContent: () => (
        <>
          <p className="sample-class-name">Sample JSX</p>
        </>
      ),
    };
    renderer = ReactShallowRenderer.createRenderer();
  });

  afterEach(() => {
    renderer.unmount();
  });

  it('should construct', () => {
    const settingSection = renderer.render(<SettingsSection {...minimal} />);
    expect(settingSection).toMatchSnapshot();
  });

  it('should have child section when expanded', () => {
    const props: SettingSectionComponentProps = { ...minimal, expanded: true };
    renderer.render(<SettingsSection {...props} />);
    const settingSection = renderer.getRenderOutput();

    expect(settingSection.props.className).toBe('msla-setting-section');

    const settingSectionChild = settingSection.props.children;
    expect(settingSectionChild.props.className).toBe('msla-setting-section-content');
  });
});
