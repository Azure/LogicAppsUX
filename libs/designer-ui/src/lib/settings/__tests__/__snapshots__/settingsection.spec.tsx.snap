// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/settings/settingsection should construct 1`] = `
<div
  className="msla-setting-section"
>
  <div
    className="msla-setting-section-content"
  >
    <div
      className="msla-setting-section-header"
      onClick={[Function]}
    >
      <CustomizedIconButton
        ariaLabel="Expand random setting"
        className="msla-setting-section-header-icon"
        iconProps={
          Object {
            "iconName": "ChevronRightMed",
            "styles": Object {
              "root": Object {
                "color": "#514f4e",
                "fontSize": 14,
              },
            },
          }
        }
        styles={
          Object {
            "icon": Object {
              "color": "#8a8886",
              "fontSize": 12,
            },
          }
        }
      />
      <button
        className="msla-setting-section-header-text"
      >
        random setting
      </button>
    </div>
    <Separator
      className="msla-setting-section-separator"
      styles={
        Object {
          "root": Object {
            "color": "#eaeaea",
          },
        }
      }
    />
  </div>
</div>
`;
